BH-1078: Add Taxonomy List

nickcernis opened the pull request:
- Implements the taxonomy table on the taxonomies developer app view.
- Adds the taxonomy options button and drop down.
- Adds “Edit” and “Delete” options to the drop down. 

### Notes
- Edit and Delete options are wired up to work but will show a TODO alert for now. (This targets the 'taxonomies' branch, so we won't release it until this is fixed in separate PRs.)
- We decided in Slack to remove the Bulk Action option for now.
- I cleared the column heading changes with Mason. (The [Figma mockup](https://www.figma.com/file/6b3kxG7r2OTv2A024CQiUU/Taxonomy) has different headings.)

### To test
1. Click “View Taxonomies from the main model page or visit `/wp-admin/?page=atlas-content-modeler&view=taxonomies` or 
2. Add one or more taxonomies, which should appear in the table.
3. Try the “Edit” or “Delete” actions, which should present you with an alert.

### Before
<img width="1229" alt="Screenshot 2021-07-01 at 18 51 23" src="https://user-images.githubusercontent.com/647669/124161175-612d2000-da9d-11eb-8145-7c9e609a832b.png">

### After
<img width="1207" alt="Screenshot 2021-07-01 at 18 04 52" src="https://user-images.githubusercontent.com/647669/124155566-e7923380-da96-11eb-86ce-30db4746c294.png">

nickcernis committed:
refactor: use more general names for color vars

These need to be used by the taxonomies view on
elements that are not disabled.

nickcernis committed:
style: add bootstrap tables and override colors

nickcernis committed:
feat: add taxonomies table component

nickcernis committed:
style: add taxonomies view custom styling

For styles that are more easily set with SCSS than
with bootstrap classes.

nickcernis committed:
feat: add TaxonomiesDropdown component

nickcernis committed:
feat: include taxonomies options dropdown

nickcernis committed:
fix: improve checkbox labels

nickcernis committed:
style: improve taxonomies option button styling

nickcernis committed:
style: improve taxonomies option button alignment

nickcernis committed:
feat: add “bulk action” taxonomy controls

nickcernis committed:
style: improve bulk actions control styling

nickcernis committed:
revert: remove bulk actions for now

After discussion with @mindctrl we plan to defer
bulk actions to implement later.

nickcernis committed:
chore: remove unused import

nickcernis committed:
style: fix table width to meet at the right edge

nickcernis committed:
test: confirm that the model appears in the table

ChrisWiegman reviewed (commented):
Can you please better describe the testing you're doing on this? It looks like there are acceptance tests. Is that all?

I also don't think we should require a model be selected. This might make creating taxonomies easier in that they can be assigned later and opens us up to a better/v2 workflow of allowing a user to pick a taxonomy from the model screen. (screenshot of where I'm talking about is below)

![Screen Shot 2021-07-01 at 12 25 16 PM](https://user-images.githubusercontent.com/394675/124158521-05946f80-da68-11eb-9adb-6c0f6db2f097.png)

nickcernis commented:
Hi @ChrisWiegman! Thanks for testing it out. I think you may have missed https://github.com/wpengine/atlas-content-modeler/pull/183, which implemented the form, submission logic, and associated tests you mention. This PR just adds a table where a bare list of models used to display.

I updated the first comment above with a before and after screenshot in case that helps.

> Can you please better describe the testing you're doing on this?

This PR only changes output of the model list from showing a list of plural names to showing a table. The change to the e2e test I made reflects that this was the only testable change.

> It looks like there are acceptance tests. Is that all?

Tests for the dependent REST endpoints and the e2e tests for the form functionality were added in https://github.com/wpengine/atlas-content-modeler/pull/183. We should of course also add tests for the upcoming taxonomy edit/delete functionality in the PRs for those features.

> I also don't think we should require a model be selected. 

This was part of the acceptance criteria for https://github.com/wpengine/atlas-content-modeler/pull/183 / [https://wpengine.atlassian.net/browse/BH-1075](BH-1075):

> User can associate the taxonomy with one or more models during creation.

(One or more, not zero or more.) Happy to chat about the implications and remove that limitation in a separate PR if we need to. I wonder if there's a risk of someone skipping the model list and wondering why their taxonomy isn't in use anywhere, though? We need to put some thought into this before undoing it.

nickcernis committed:
fix: add empty taxonomy table state

nickcernis committed:
test: check for empty taxonomies table message

nickcernis commented:
Updated with an empty table state and test for the empty state.

I added a [Jira story](https://wpengine.atlassian.net/browse/BH-1111) to make a PR template to prompt to include context about tests in future PRs. Thanks for the feedback, @ChrisWiegman. ❤️

ChrisWiegman reviewed (approved):
Thank you for the answers Nick. Indeed I had not reviewed #183 so the context of this makes much more sense now. Approving.

apmatthews reviewed (approved):
LGTM! Down the road we'll probably want to extract a more generic version of the dropdown component that can be shared between models and taxonomies, but I actually like having two fully coded use cases as a starting point for that kind of extraction. It makes it easier to see what props a generic version might need and where different responsibilities should lie.

nickcernis commented:
Thanks for the reviews, all!

> Down the road we'll probably want to extract a more generic version of the dropdown component that can be shared between models and taxonomies…

💯 Completely agree. We should review the places we're using a modal and abstract that too. I'll add backlog items for both.
